src/Chunk.ts

```diff
@@ -390,8 +390,8 @@ export default class Chunk {
 		if (!es) {
 			if (format === 'cjs') {
 				dynamicImportMechanism = {
-					left: 'Promise.resolve(require(',
-					right: '))'
+					left: 'Promise.resolve({ default: require(',
+					right: ') })'
```

Comment:
What is the scenario you hope to fix here? Test? As you can see with the test below, this makes named exports no longer work (and again shows that we probably need functional tests). My feeling is that the correct approach would be more complicated.

Reply:
Great catch. Ok so the rule is this:

* When the dynamic import is an external module, we need to treat it as `{ default: module }` as that is how interop works when importing CommonJS from ES modules.
* When the dynamic import is a known module from our current bundle operation, we treat it as a full module object, as it's effectively within our private interface for bundling.

I've added the adjustments to make these cases work out now.